packages/app/src/app/hooks/useDismissible.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffects } from 'app/overmind';
+
+/**
+ * Localstorage keys used for dismissible modals and banners.
+ */
+type DismissibleKeys =
+  // TODO - Any key in this type should refer to a dismissible item. Right
+  // now there are none yet.
+  'TODO';
+
+type Dismissibles = Partial<Record<DismissibleKeys, true>>;
+
+export const useDismissible = () => {
+  const { browser } = useEffects();
+  const dismissibles = browser.storage.get<Dismissibles>('DISMISSIBLES');
+
+  const isDismissed = (key: DismissibleKeys) => {
+    return dismissibles?.[key] === true;
+  };
+
+  const dismiss = (key: DismissibleKeys) => {
+    const prevDismissibles = dismissibles || {};
+
+    browser.storage.set('DISMISSIBLES', {
+      ...prevDismissibles,
+      [key]: true,
+    });
+  };
+
+  return { isDismissed, dismiss };
```

Comment:
If each instance tracks per key, could the return be similar to React's own `useState`?

```suggestion
  return [isDismissed, dismiss];
```

This way we don't need to rename keys.